### PR TITLE
Fix/webengine close connection resource constraint

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
This PR is **ready** for review.

### Summary
Refactoring : use setters in appHMI type boolean bvalues
Fix : remove connection on resource constrains for all application

### CODE REVIEW

File: src/components/application_manager/src/application_impl.cc [hunk 1]
```diff
@@ -246,9 +247,9 @@ bool ApplicationImpl::is_voice_communication_supported() const {
   return is_voice_communication_application_;
 }
 
-void ApplicationImpl::set_voice_communication_supported(
-    bool is_voice_communication_supported) {
-  is_voice_communication_application_ = is_voice_communication_supported;
+void ApplicationImpl::set_voice_communication_supported(bool option) {
+  LOG4CXX_TRACE(logger_, "option " << option);
```
Comment (IGapchuk) [inline src/components/application_manager/src/application_impl.cc:251]: @LuxoftAKutsan Maybe it makes sense to add std::boolalpha into the output stream for more readable log format?

File: src/components/application_manager/src/application_impl.cc [hunk 2]
```diff
@@ -203,7 +203,7 @@ void ApplicationImpl::ChangeSupportingAppHMIType() {
   is_navi_ = false;
   is_voice_communication_application_ = false;
   mobile_projection_enabled_ = false;
-  webengine_projection_enabled_ = false;
+  set_webengine_projection_enabled(false);
   const smart_objects::SmartObject& array_app_types = *app_types_;
```
Comment (SNiukalov) [inline src/components/application_manager/src/application_impl.cc:207]: Different approaches, my suggestion:
set_is_navi(false);
set_voice_communication_supported(false);
set_mobile_projection_enabled(false);

Comment (LuxoftAKutsan) [reply src/components/application_manager/src/application_impl.cc:207]: Enhancement done 35c19b0360

File: src/components/application_manager/src/application_impl.cc (hunk 1 above)
Comment (LuxoftAKutsan) [reply src/components/application_manager/src/application_impl.cc]: Enhancement done 35c19b0360

**Pull request merged**